base/MAPL_SwathGridFactory.F90

```diff
@@ -0,0 +1,1259 @@
+      procedure :: generate_file_reference3D
+      procedure :: decomps_are_equal
+      procedure :: physical_params_are_equal
+   end type SwathGridFactory
+
+   character(len=*), parameter :: MOD_NAME = 'MAPL_SwathGridFactory::'
+
+   interface SwathGridFactory
+      module procedure SwathGridFactory_from_parameters
+   end interface SwathGridFactory
+
+   interface set_with_default
+      module procedure set_with_default_integer
+      module procedure set_with_default_real
+      module procedure set_with_default_real64
+      module procedure set_with_default_character
+      module procedure set_with_default_bounds
+   end interface set_with_default
+   
+contains
+
+
+   function SwathGridFactory_from_parameters(unusable, grid_name, &
+        & im_world, jm_world, lm, nx, ny, ims, jms, rc) result(factory)
+      type (SwathGridFactory) :: factory
+      class (KeywordEnforcer), optional, intent(in) :: unusable
+      character(len=*), optional, intent(in) :: grid_name
+
+      ! grid details:
+      integer, optional, intent(in) :: im_world
+      integer, optional, intent(in) :: jm_world
+      integer, optional, intent(in) :: lm
+      
+      ! decomposition:
+      integer, optional, intent(in) :: nx
+      integer, optional, intent(in) :: ny
+      integer, optional, intent(in) :: ims(:)
+      integer, optional, intent(in) :: jms(:)
+
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+
+      _UNUSED_DUMMY(unusable)
+
+      STOP 'Stop:  SwathGridFactory_from_parameters is not tested'
+      
+      call set_with_default(factory%grid_name, grid_name, MAPL_GRID_NAME_DEFAULT)
+      call set_with_default(factory%nx, nx, MAPL_UNDEFINED_INTEGER)
+      call set_with_default(factory%ny, ny, MAPL_UNDEFINED_INTEGER)
+      call set_with_default(factory%im_world, im_world, MAPL_UNDEFINED_INTEGER)
+      call set_with_default(factory%jm_world, jm_world, MAPL_UNDEFINED_INTEGER)
+      call set_with_default(factory%lm, lm, MAPL_UNDEFINED_INTEGER)
+
+      ! default is unallocated
+      if (present(ims)) factory%ims = ims
+      if (present(jms)) factory%jms = jms
+
+!      call set_with_default(factory%lon_range, lon_range, RealMinMax(MAPL_UNDEFINED_REAL,MAPL_UNDEFINED_REAL))
+!      call set_with_default(factory%lat_range, lat_range, RealMinMax(MAPL_UNDEFINED_REAL,MAPL_UNDEFINED_REAL))
+!      call set_with_default(factory%force_decomposition, force_decomposition, .false.)
+
+      call factory%check_and_fill_consistency(rc=status)
+      _VERIFY(status)
+
+      _RETURN(_SUCCESS)
+
+   end function SwathGridFactory_from_parameters
+
+
+   function make_new_grid(this, unusable, rc) result(grid)
+      type (ESMF_Grid) :: grid
+      class (SwathGridFactory), intent(in) :: this
+      class (KeywordEnforcer), optional, intent(in) :: unusable
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+
+      _UNUSED_DUMMY(unusable)
+      grid = this%create_basic_grid(rc=status)
+      _VERIFY(status)
```

Comment:
```suggestion
      grid = this%create_basic_grid(_RC)
```
Please switch to this style throughout.

Reply:
Thanks. I am changing to this style throughout the code.